### ISSUE TITLE
Getting Error: UUID '...' is not a valid UUID

### ISSUE THREAD

**Issue by funkflute** [sibartlett/homebridge-wink3#29]
I've been getting this error trying to connect to my Quirky Aros - any ideas what's going on?

```
[2017-8-2 23:49:35] [Wink] Refreshing devices...
[2017-8-2 23:49:39] [Wink] Failed to refresh devices. Error: UUID '[removed]' is not a valid UUID. Try using the provided 'generateUUID' function to create a valid UUID from any arbitrary string, like a serial number.
    at new PlatformAccessory (/opt/node/lib/node_modules/homebridge/lib/platformAccessory.js:17:34)
    at WinkPlatform.addDevice (/opt/node/lib/node_modules/homebridge-wink3/src/WinkPlatform.js:142:23)
    at Array.forEach (native)
    at /opt/node/lib/node_modules/homebridge-wink3/src/WinkPlatform.js:181:13
    at Generator.next (<anonymous>)
    at step (/opt/node/lib/node_modules/homebridge-wink3/dist/WinkPlatform.js:52:191)
    at /opt/node/lib/node_modules/homebridge-wink3/dist/WinkPlatform.js:52:361
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

```

**Comment by sibartlett** [comment]
I will publish a new version soon that you can try, sorry for the delay!

**Comment by sibartlett** [comment]
I've published a new version (1.5.1), can you see if that works for you?

**Comment by funkflute** [comment]
Unfortunately, same error with v1.5.2.
```
[2017-8-10 17:42:50] [Wink] Refreshing devices...
[2017-8-10 17:42:55] [Wink] Failed to refresh devices. Error: UUID 'XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX' is not a valid UUID. Try using the provided 'generateUUID' function to create a valid UUID from any arbitrary string, like a serial number.
    at new PlatformAccessory (/opt/node/lib/node_modules/homebridge/lib/platformAccessory.js:17:34)
    at WinkPlatform.addDevice (/opt/node/lib/node_modules/homebridge-wink3/src/WinkPlatform.js:144:23)
    at Array.forEach (native)
    at /opt/node/lib/node_modules/homebridge-wink3/src/WinkPlatform.js:184:13
    at Generator.next (<anonymous>)
    at step (/opt/node/lib/node_modules/homebridge-wink3/dist/WinkPlatform.js:52:191)
    at /opt/node/lib/node_modules/homebridge-wink3/dist/WinkPlatform.js:52:361
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

**Comment by sibartlett** [comment]
Does the UUID contain any letters other than A to F?

**Comment by funkflute** [comment]
it's all hexadecimal with dashes in the places I marked with Xs above, like this sample UUID:
`c264a30d-c995-4377-b01d-c6887c81446a`

**Comment by sibartlett** [comment]
homebridge uses a regular expression to validate the UUID.

Are you familiar with regexes? If so can you check your UUID against this [regex](https://github.com/KhaosT/HAP-NodeJS/blob/2ea9d761d9bd7593dd1949fec621ab085af5e567/lib/util/uuid.js#L28)? Otherwise, you're welcome to email me the actual UUID.

Would be good to know the actual reason why it fails validation.

**Comment by funkflute** [comment]
Sent yo your emails. Looks like it’s failing because the 3 set of numbers starts with 0, not [1-5] and the 4th set doesn’t start with [89ab]

**Comment by sibartlett** [comment]
Seems the issue is that Wink's UUIDs do not conform to RFC 4122.

I'll update the code, so that I explicitly check if Wink's UUID is valid, otherwise generate a new UUID.

**Comment by sibartlett** [comment]
I've published a new version (1.5.4) which I hope fixes this issue.

**Comment by funkflute** [comment]
Now I'm getting this: **TypeError: uuid.generateUUID is not a function**
```
Failed to refresh devices. TypeError: uuid.generateUUID is not a function
    at WinkPlatform.addDevice (/opt/node/lib/node_modules/homebridge-wink3/src/WinkPlatform.js:147:14)
    at Array.forEach (native)
    at /opt/node/lib/node_modules/homebridge-wink3/src/WinkPlatform.js:188:13
    at Generator.next (<anonymous>)
    at step (/opt/node/lib/node_modules/homebridge-wink3/dist/WinkPlatform.js:52:191)
    at /opt/node/lib/node_modules/homebridge-wink3/dist/WinkPlatform.js:52:361
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)```

**Comment by sibartlett** [comment]
Sorry, I had a typo there. I just released another version (1.5.5) for you to try.

**Comment by funkflute** [comment]
YoUUU got it! Working now! Thanks so much for your work.
